prow/cmd/generic-autobumper/main.go

```diff
@@ -19,47 +19,467 @@ package main
+	// List of prefixes that the autobumped is looking for, and other information needed to bump them. Must have at least 1 prefix.
+	Prefixes []Prefix `yaml:"prefixes"`
+}
+
+// Prefix is the information needed for each prefix being bumped.
+type Prefix struct {
+	// Name of the tool being bumped
+	Name string `yaml:"name"`
+	// The image prefix that the autobumper should look for
+	Prefix string `yaml:"prefix"`
+	// File that is looked at to determine current upstream image when bumping to upstream. Required only if targetVersion is "upstream"
+	RefConfigFile string `yaml:"refConfigFile"`
+	// File that is looked at to determine current upstream staging image when bumping to upstream staging. Required only if targetVersion is "upstream-staging"
+	StagingRefConfigFile string `yaml:"stagingRefConfigFile"`
+	// The repo where the image source resides for the images with this prefix. Used to create the links to see comparisons between images in the PR summary.
+	Repo string `yaml:"repo"`
+	// Whether or not the format of the PR summary for this prefix should be summarised.
+	Summarise bool `yaml:"summarise"`
+	// Whether the prefix tags should be consistent after the bump
+	ConsistentImages bool `yaml:"consistentImages"`
+}
+
+func parseOptions() (*Options, *bumper.Options, error) {
 	var config string
 	var labelsOverride []string
+	var skipPullRequest bool
 
 	flag.StringVar(&config, "config", "", "The path to the config file for the autobumber.")
 	flag.StringSliceVar(&labelsOverride, "labels-override", nil, "Override labels to be added to PR.")
+	flag.BoolVar(&skipPullRequest, "skip-pullrequest", false, "")
 	flag.Parse()
 
-	var o bumper.Options
+	var o Options
+	var pro bumper.Options
 	data, err := ioutil.ReadFile(config)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to read in config file, %s", config)
+		return nil, nil, fmt.Errorf("Failed to read in config file, %s", config)
+	}
+
+	err = yaml.Unmarshal(data, &o)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to parse yaml file, %s", err)
 	}
 
-	if err := yaml.UnmarshalStrict(data, &o); err != nil {
-		return nil, fmt.Errorf("Failed to parse yaml file, %s", err)
+	if err := yaml.Unmarshal(data, &pro); err != nil {
+		return nil, nil, fmt.Errorf("Failed to parse yaml file, %s", err)
 	}
 
 	if labelsOverride != nil {
-		o.Labels = labelsOverride
+		pro.Labels = labelsOverride
 	}
-	return &o, nil
+	pro.SkipPullRequest = skipPullRequest
+	return &o, &pro, nil
+}
+
+func validateOptions(o *Options) error {
+	if len(o.Prefixes) == 0 {
+		return fmt.Errorf("Must have at least one Prefix specified")
+	}
+	if len(o.IncludedConfigPaths) == 0 {
+		return fmt.Errorf("includedConfigPaths is mandatory")
+	}
+	if o.TargetVersion != latestVersion && o.TargetVersion != upstreamVersion &&
+		o.TargetVersion != upstreamStagingVersion && !tagRegexp.MatchString(o.TargetVersion) {
+		logrus.Warnf("Warning: targetVersion is not one of %v so it might not work properly.",
```

Comment:
I rarely use formatted strings with logrus. Using fields is the preferred style.

Reply:
done